Content/Editor/MaterialTemplates/Features/ForwardShading.hlsl

```diff
@@ -67,7 +67,8 @@ void PS_Forward(
 	gBuffer.Color = material.Color;
 	gBuffer.Specular = material.Specular;
 	gBuffer.AO = material.AO;
-	gBuffer.ViewPos = mul(float4(materialInput.WorldPosition, 1), ViewMatrix).xyz;
+	// gBuffer.ViewPos is the view position in WORLD SPACE
+	gBuffer.ViewPos = ViewPos;
```

Comment:
That's incorrect. `GBufferSample.ViewPos` is the view-space location of the pixel

Reply:
Oh, my fault, I've messed up GBufferSample with GBufferData. It's really confusing, I'd like to add some additional comments to demonstrate this.